### PR TITLE
Improve domain actions permissions with new api properties

### PR DESCRIPTION
**Pull request by gius80**
## Changes proposed in this Pull Request

This PR aims to add some additional checks to edit domain contact informations, dns records and name servers, using new domain property added in this diff D76803-code

dispute-flag-settings

dispute-flag-dns

Schermata 2022-03-16 alle 10 23 36

## Testing instructions

- Apply the back-end patch D76803-code
- Build this branch locally or open the live Calypso link
- Set a dispute flag on a test domain
- Open domain settings page for that domain and verify that _Name servers_, _DNS records_ and _Contact Information_ cards show a warning message
- Verify that _DNS page_ (`domains/manage/all/{:domain}/dns/{:site}`) shows a warning message
- Verify that _Edit Contact Info page_ (`domains/manage/all/{:domain}/edit-contact-info/{:site}`) shows a warning message

### CODE REVIEW

File: client/my-sites/domains/domain-management/components/domain/cannot-update-contact-info.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+}: {
+	redesigned: boolean;
+	domain: ResponseDomain;
+} ) => {
+	return (
+		<InfoNotice redesigned={ redesigned } text={ domain.cannotUpdateContactInfoReason as string } />
+	);
+};
+
+export default CannotUpdateContactInfo;
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/components/domain/cannot-update-contact-info.tsx]: `CannotUpdateContactInfo` and `CannotManageDnsRecords` are the same and basically wrappers around `InfoNotice`, so I think maybe we should use `InfoNotice` directly in place of them 🙂

**Review by leonardost — CHANGES_REQUESTED** [review]
Tested and LGTM! There's only one comment that I think should be addressed before we ship this, but overall looks great!

File: client/my-sites/domains/domain-management/components/domain/cannot-update-contact-info.tsx (hunk 1 above)
Comment (gius80) [reply client/my-sites/domains/domain-management/components/domain/cannot-update-contact-info.tsx]: You're totally right! I've refactored the code 😉

**Review by delputnam — APPROVED** [review]
Nice job.   LGTM.

**Pull request merged**